fluentd/vendored_gem_src/syslog_protocol/lib/syslog_protocol/packet.rb

```diff
@@ -112,6 +113,55 @@ def hostname=(h)
       @hostname = h
     end
 
+    def appname=(a)
+      unless a and a.is_a? String and a.length > 0
+        raise ArgumentError.new("Appname may not be omitted")
+      end
+      #if a.length > 48
+      if a.length > 500
+        raise ArgumentError, "Appname must not be longer than 48 characters"
+      end
+      if a =~ /\s/
```

Comment:
Not certain if the regex captures all whatespace but may be more descriptive to use `include?` https://ruby-doc.org/core-2.5.1/String.html#method-i-include-3F

Reply:
The next regex after this checks for allowed values as per rfc. So this already excludes space (`0x20`).
Perhaps `/\s/` regex's intention is not to check for all whitespace characters but check for a common case of `' '` character. This check is actually not needed.

```
      if a =~ /[^\x21-\x7E]/      
        raise ArgumentError.new("Appname may only contain ASCII characters 33-126")      
      end      
```